File: ruby/bob/bob.rb
require 'pry'
class Bob
  def hey(remark)
    case
      when yelling?(remark) then "Whoa, chill out!"
      when question?(remark) then "Sure."
      when silent?(remark) then "Fine. Be that way!"
      else "Whatever."
    end
  end

  def yelling?(remark)
    remark == remark.upcase && remark != remark.downcase
  end

  def question?(remark)
    remark.end_with?("?")
  end

  def silent?(remark)
    remark.strip.empty?
  end
end

File: javascript/bob/bob.js
var Bob = function() {};

Bob.prototype.hey = function(input) {
  if (isUpperCase(input)) {
    return 'Whoa, chill out!';
  }
  if (isQuestion(input)) {
    return 'Sure.';
  }
  if (silentTreatment(input)) {
    return 'Fine. Be that way!';
  }
    return 'Whatever.';
};

function isUpperCase(input) {
  return input === input.toUpperCase() &&
  input.toUpperCase() !== input.toLowerCase();
}

function isQuestion(input) {
  return input.endsWith('?');
}

function silentTreatment(input) {
  return input.trim() === '';
}

module.exports = Bob;
